### PR TITLE
Include track flags

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/ride/track.c [hunk 1]
```diff
@@ -818,15 +818,15 @@ static money32 track_place(int rideIndex, int type, int originX, int originY, in
...
+			if (FlatTrackFlags[type] & TRACK_ELEM_FLAG_0200) {
 				if (gTrackGroundFlags & TRACK_ELEMENT_LOCATION_IS_UNDERGROUND) {
 					gGameCommandErrorText = STR_CAN_ONLY_BUILD_THIS_ABOVE_GROUND;
 					return MONEY32_UNDEFINED;
 				}
 			}
 		}
 		else {
-			if (RCT2_ADDRESS(0x0099423C, uint16)[type] & 0x200) {
+			if (TrackFlags[type] & TRACK_ELEM_FLAG_0200) {
```
Comment (marijnvdwerf) [inline src/ride/track.c]: > Seems to be a flag to mark a piece as only suitable for above ground.
> -- @Gymnasiast (https://github.com/OpenRCT2/OpenRCT2/pull/4181#discussion_r73047269)

File: src/ride/track.c [hunk 2]
```diff
@@ -835,15 +835,15 @@ static money32 track_place(int rideIndex, int type, int originX, int originY, in
...
+			if (FlatTrackFlags[type] & TRACK_ELEM_FLAG_0001) {
 				if (!(gMapGroundFlags & ELEMENT_IS_UNDERWATER)) {
 					gGameCommandErrorText = STR_CAN_ONLY_BUILD_THIS_UNDERWATER;
 					return MONEY32_UNDEFINED;
 				}
 			}
 		}
 		else {
-			if (RCT2_ADDRESS(0x0099423C, uint16)[type] & 1) {
+			if (TrackFlags[type] & TRACK_ELEM_FLAG_0001) { // No element has this flag
```
Comment (marijnvdwerf) [inline src/ride/track.c]: > Seems to be a flag to mark a piece as only suitable to be built underwater.
> -- @Gymnasiast (https://github.com/OpenRCT2/OpenRCT2/pull/4181#discussion_r73047565)

**Comment by marijnvdwerf** [comment]
`FlatTrackFlags` contain a lot of irrelevant data, as only a few flat track types exist.

**Comment by IntelOrca** [comment]
Is this ready?

**Comment by marijnvdwerf** [comment]
There's the enum naming thing, and I don't know what to do with the garbage data for the FlatTracks

**Comment by IntelOrca** [comment]
Needs rebasing

**Pull request merged**